app/public/static/js/components/ordered-puzzle.js

```diff
@@ -0,0 +1,196 @@
+          let textBoxProp = JSON.parse(JSON.stringify(data.images[index]));
+          textBoxProp.color = "yellow";
+          textBoxProp.width = puzzlePieceWidth;
+          textBoxProp.height = puzzlePieceHeight;
+          textBoxProp.xTarget = textBoxProp.xTarget * puzzlePieceWidth;
+          textBoxProp.yTarget = textBoxProp.yTarget * puzzlePieceHeight;
+
+          let target = document.createElement("a-entity");
+          target.setAttribute("id", "puzzle-target-" + index);
+          target.setAttribute(
+            "target-box",
+            "jsonData",
+            JSON.stringify(textBoxProp)
+          );
+          el.appendChild(target);
+
+          // Set image's target location
+          puzzlePiece.setAttribute(
+            "xTarget",
+            data.images[index].xTarget * puzzlePieceWidth
+          );
+          puzzlePiece.setAttribute(
+            "yTarget",
+            data.images[index].yTarget * puzzlePieceHeight
+          );
+          puzzlePiece.setAttribute("onTarget", false);
+
+          puzzlePiece.addEventListener("dragend", function (event) {
+            event.target.setAttribute("onTarget", isOnTarget(event.target));
+            if (isPuzzleComplete(puzzlePieceCache)) {
+              // emit event to update state
+              el.sceneEl.emit("solvedObject", { id: data.id });
+              // emit event to close popup (after 1 second for now) - TODO: disabled for now
+              // need designer input on whether we should autoclose this
+              // el.sceneEl.emit("dcc-success-close-popup", { seconds: 1 });
+              // emit event if it is last object, to indicate scene is solved
+              if (is_last_object) {
+                el.sceneEl.emit("dcc-success-scene-complete");
+              }
+            }
+          });
+        }
+
+        el.appendChild(puzzlePiece);
+      };
+      rawImageEl.src = data.images[i].imageSrc;
+    }
+
+    if (data.useTargets) {
+      // Once window is loaded, calculated if current puzzle pieces are on their target.
+      window.addEventListener("load", function () {
+        puzzlePieceCache.forEach((puzzlePiece) => {
+          puzzlePiece.setAttribute("onTarget", isOnTarget(puzzlePiece));
+        });
+      });
+    }
+  },
+  update: function () {
+    this.puzzleIsSolved = this.data.isSolved[this.id];
+
+    if (typeof this.puzzleIsSolved === "undefined") {
+      // not loaded yet, do nothing
+    } else if (this.puzzleIsSolved === true && this.useTargets) {
+      // Already solved
+      this.solvedPuzzleEntity.setAttribute("visible", "true");
+    } else {
+      // not solved yet, do nothing for now
+    }
+  },
+});
+
+function isOnTarget(puzzlePiece) {
+  const delta = puzzlePiece.object3D.children[0].position;
+  const originalPos = puzzlePiece.getAttribute("position");
+  const targetPos = {
+    x: puzzlePiece.getAttribute("xTarget"),
+    y: puzzlePiece.getAttribute("yTarget"),
+  };
+  const accuracyReq = 1;
```

Comment:
Can we change this to something like?
```
const accuracyReq = scaleBy/3;
```

Otherwise at lower scaling I am able get solved state in such situations:
![image](https://user-images.githubusercontent.com/35475783/116796752-1565f680-aaad-11eb-9bfb-af2a974b4517.png)